packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -791,11 +791,16 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
     }
 
     Rect? targetRect;
-    if (targetRenderObject != null && targetRenderObject != object) {
-      targetRect = MatrixUtils.transformRect(
-        targetRenderObject.getTransformTo(object),
-        object.paintBounds.intersect(targetRenderObject.paintBounds),
-      );
+    try {
+      if (targetRenderObject != null && targetRenderObject != object) {
+        targetRect = MatrixUtils.transformRect(
+          targetRenderObject.getTransformTo(object),
+          object.paintBounds.intersect(targetRenderObject.paintBounds),
+        );
+      }
+    } catch (e) {
+      // If the transform cannot be computed, return.
+      return;
```

Comment:
It's better than crashing, but it still isn't ideal that `ensureVisible` would just give up if it's in an OverlayPortal? I think I had tried to fix this (https://github.com/flutter/flutter/pull/130192). Let me reopen the PR and see if that fixes this.

Reply:
Turned into a draft for now.

Reply:
https://github.com/flutter/flutter/pull/148897. I think the test added in the PR also passes with the patch.